src/meta_merge/core.cljc

```diff
@@ -1,11 +1,16 @@
 (ns meta-merge.core
   (:require [clojure.set :as set]))
 
+(defn metadatable?
```

Comment:
Right, but the protocols aren't connected. So you could potentially satisfy `IMeta` without `IWithMeta` or vice versa. This means that the checks for `meta*` and `with-meta*` are going to be different:

```clojure
(defn- meta* [obj]
  (if #?(:clj  (instance? clojure.lang.IObj obj)
         :cljs (satisfies? IMeta obj))
    (meta obj)
    nil))

(defn with-meta* [obj m]
  (if #?(:clj  (instance? clojure.lang.IObj obj)
         :cljs (satisfies? IWithMeta obj))
    (with-meta obj m)
    nil))
```

Notice how the conditions are only identical for Clojure; for ClojureScript we need to check separate protocols.